### DISCUSSION TITLE
Safelist with Attributify

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [windicss/windicss#613] [Q&A] [answered]
Hello,

I was working on a Vuepress project with `attributify` mode enabled and I created a small `icon` component to reuse it multiple times.

It's like this:

```vue
<template>
  <div :class = "`h-${$props.size} w-${$props.size}`">
    <svg xmlns = "http://www.w3.org/2000/svg" viewBox = "0 0 24 24" w:svg = "fill-none stroke-2 stroke-cap-round stroke-current stroke-join-round">
      <path :d = "$props.path"/>
    </svg>
  </div>
</template>
<script>
  export default {
    name: 'Icon',
    props: {
      path: String,
      size: Number
    }
  }
</script>
```

It's working fine this way, but previously, I had it like:

```vue
<div :w:h = "$props.size" :w:w = "$props.size">
// rest same as above
```

Since this can't be directly extracted as a class by WindiCSS, I added it to the safelist like:

```js
{
  safelist: 'h-5 w-5'
}
```

I used 5 since that's going to be the default size for most icons. But this config didn't work until I used the `class` attribute as shown at first.

So, my question is that, is there any way to specify safelist for `attributify` mode? This would just help me keep the code consistent and use `class` attribute only for custom classes.

I tried this:

```js
{
  safelist: '[w:h~="5"] [w:w~="5"]'
}
```

But that doesn't seem to work.

**Comment by antfu** [comment, marked as answer]
Yeah that's a bit tricky to support. Currently as a workaround you can use comment somewhere in your files to serve the same purpose as safelist:

```html
<!-- safelist.vue -->

<!--
<div w:h="5" w:w="5">
-->
```

**Reply by hrishikesh-k** [reply]
Oh yeah that's a good workaround too. Thanks!
